pkg/installmanager/installmanager.go

```diff
@@ -864,6 +865,58 @@ func (m *InstallManager) generateAssets(cd *hivev1.ClusterDeployment, workerMach
 		}
 	}
 
+	// Day 0: Add MachinePool label to MachineSet manifests
+
+	mapPoolsByType := map[string]string{}
+	mpL := &hivev1.MachinePoolList{}
+	if err := m.DynamicClient.List(context.Background(), mpL, &client.ListOptions{Namespace: m.Namespace}); err != nil {
+		return err
+	}
+	for _, pool := range mpL.Items {
+		if pool.Spec.ClusterDeploymentRef.Name == cd.Name {
+			mapPoolsByType[pool.Spec.Name] = pool.Name
+		}
+	}
+
+	if len(mapPoolsByType) > 0 {
+		// TODO: Handle error arg to WalkDirFunc
+		err = filepath.WalkDir(filepath.Join(m.WorkDir, "openshift"), func(path string, d fs.DirEntry, _ error) error {
+			if d.IsDir() || (filepath.Ext(path) != ".yaml" && filepath.Ext(path) != ".yml") {
+				return nil
+			}
+
+			manifestBytes, err := os.ReadFile(path)
+			if err != nil {
+				m.log.WithError(err).Error("error reading manifest file")
+				return err
+			}
+
+			modifiedManifestBytes, err := patchLabelMachineSetManifest(manifestBytes, mapPoolsByType, m.log)
+			if err != nil {
+				m.log.WithError(err).Error("error patching worker machineset manifest")
+				return err
+			}
+			if modifiedManifestBytes != nil {
+				info, err := d.Info()
+				if err != nil {
+					m.log.WithError(err).Error("error retrieving file info")
+					return err
+				}
+				err = os.WriteFile(path, *modifiedManifestBytes, info.Mode())
+				if err != nil {
+					m.log.WithError(err).Error("error writing manifest")
+					return err
+				}
+			}
+
+			return nil
+		})
+		if err != nil {
+			m.log.WithError(err).Error("error finding machine pool manifests")
+			return err
+		}
+	}
```

Comment:
Yeah, let's totally DRY this section with the next. With very little refactoring, we can correctly call both `patch*` funcs from within a single `WalkDir` looper.